src/Features/CSharp/Portable/BraceCompletion/CurlyBraceCompletionService.cs

```diff
@@ -0,0 +1,416 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Composition;
+using System.Diagnostics;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.BraceCompletion;
+using Microsoft.CodeAnalysis.CSharp.Formatting;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Formatting;
+using Microsoft.CodeAnalysis.Formatting.Rules;
+using Microsoft.CodeAnalysis.Host.Mef;
+using Microsoft.CodeAnalysis.Indentation;
+using Microsoft.CodeAnalysis.Options;
+using Microsoft.CodeAnalysis.PooledObjects;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+using Microsoft.CodeAnalysis.Text;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.CSharp.BraceCompletion
+{
+    [Export(LanguageNames.CSharp, typeof(IBraceCompletionService)), Shared]
+    internal class CurlyBraceCompletionService : AbstractBraceCompletionService
+    {
+        private static readonly SyntaxAnnotation s_closingBraceSyntaxAnnotation = new("original closing brace");
```

Comment:
doc what this id for.

Reply:
also, why do we need this for culies, but not for BracketBraceCompletionService ?

Reply:
bracket brace completion doesn't do anything special afterstart / afterreturn so there isn't complexity in finding the closing brace.  But for curlies we do formatting after start and after return.  Previously it used the closing point tracking span to do find the closing brace location (to put the caret there) but those aren't available in this layer